object-cache.php

```diff
@@ -259,899 +276,969 @@ function wp_cache_flush_group( $groups = 'default' ) {
-	/**
-	 * @return array
-	 */
-	public function getNonPersistentCache() {
-		return $this->non_persistent_cache;
-	}
-
-	/**
-	 * @return array
-	 */
-	public function getNonPersistentGroups() {
-		return $this->non_persistent_groups;
-	}
-
-	/**
-	 * @return array
-	 */
-	public function getSiteVersions() {
-		return $this->site_versions;
-	}
+class WP_Object_Cache
+{
+
+    /**
+     * @var string MD5 hash of the current installation ABSPATH
+     */
+    private $abspath;
+
+    /**
+     * @var bool Stores if APCu is available.
+     */
+    private $apcu_available;
+
+    /**
+     * @var int The sites current blog ID. This only
+     *    differs if running a multi-site installations
+     */
+    private $blog_prefix;
+
+    /**
+     * @var int Keeps count of how many times the
+     *    cache was successfully received from APCu
+     */
+    public $cache_hits = 0;
+
+    /**
+     * @var int Keeps count of how many times the
+     *    cache was not successfully received from APCu
+     */
+    public $cache_misses = 0;
+
+    /**
+     * @var array Holds a list of cache groups that are
+     *    shared across all sites in a multi-site installation
+     */
+    private $global_groups = [];
+
+    /**
+     * @var array Holds an array of versions of the retrieved groups
+     */
+    private $group_versions = [];
+
+    /**
+     * @var bool True if the current installation is a multi-site
+     */
+    private $multi_site;
+
+    /**
+     * @var array Holds cache that is to be non persistent
+     */
+    private $non_persistent_cache = [];
+
+    /**
+     * @var array Holds a list of cache groups that are not to be saved to APCu
+     */
+    private $non_persistent_groups = [];
+
+    /**
+     * @var array|bool
+     */
+    private $local_cache = [];
+
+    /**
+     * @var array Holds an array of versions of the retrieved sites
+     */
+    private $site_versions = [];
+
+    private static $instance;
+
+    /**
+     * Singleton. Return instance of WP_Object_Cache
+     *
+     * @return WP_Object_Cache
+     */
+    public static function instance()
+    {
+        if (self::$instance === null) {
+            self::$instance = new WP_Object_Cache();
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * __clone not allowed
+     */
+    private function __clone()
+    {
+    }
+
+    /**
+     * Direct access to __construct not allowed.
+     */
+    private function __construct()
+    {
+        global $blog_id;
+
+        if (!defined('WP_APCU_KEY_SALT')) {
+            /**
+             * Set in config if you are using some sort of shared
+             * config where ABSPATH is the same on all sites
+             */
+            define('WP_APCU_KEY_SALT', 'wp');
+        }
+
+        /**
+         * define('WP_APCU_LOCAL_CACHE', false) to disable local
+         * array cache and force all cache to be returned from APCu
+         */
+        if (defined('WP_APCU_LOCAL_CACHE') && !WP_APCU_LOCAL_CACHE) {
```

Comment:
Using constant directly may be more efficient than using object property. PHP is able to optimize some conditions with constants and skip them completely at opcache level  (for example `if (WP_APCU_LOCAL_CACHE) { $this->something() }` can me completely removed if `WP_APCU_LOCAL_CACHE` is `false`).

Reply:
Noted I will make the appropriate changes.